packages/example-application/src/react-hook-form.tsx

```diff
@@ -0,0 +1,217 @@
+/* eslint-disable react/prop-types */
+
+import React, { useState, useCallback } from 'react';
+import * as GovUK from 'govuk-react';
+import { Link } from 'react-router-dom';
+
+import { useForm } from 'react-hook-form';
+
+const DateField = ({ input: { onChange, onBlur, ...input }, ...props }) => {
+  const [value, setValue] = useState(input.value);
+  return (
+    <GovUK.DateField
+      {...props}
+      input={{
+        onChange: (newValue) => {
+          setValue({ ...value, ...newValue });
+          onChange({ target: { value: { ...value, ...newValue }, name: input.name } });
+        },
+        onBlur: (newValue) => onBlur({ target: { value, name: input.name } }),
+        ...input,
+      }}
+    />
+  );
+};
+
+const ReactHookForm = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, submitCount },
+  } = useForm();
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const [data, setData] = useState(false);
+  const handleFormSubmit = useCallback(
+    (values) => {
+      if (isSubmitting) return;
+      setIsSubmitting(true);
+      setTimeout(() => {
```

Comment:
maybe comment here to say why this timeout is needed
// fake api call

Reply:
https://github.com/govuk-react/govuk-react/pull/916/commits/229086bb6cd0624271047a7e0703c3695fffc30e